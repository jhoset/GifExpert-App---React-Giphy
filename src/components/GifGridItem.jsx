import PropTypes from 'prop-types'

export default function GifGridItem({title, url}) {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p> {title} </p>
            <div className="button"><a href="#"> BUTTON </a></div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string
}
