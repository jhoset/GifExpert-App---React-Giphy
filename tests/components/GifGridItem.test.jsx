import { render, screen } from "@testing-library/react"
import GifGridItem from "../../src/components/GifGridItem"

describe("Tests in <GifGridItem />", () => {

    const title = 'Hamlechi';
    const url = 'https://hamlechi/photo.jpg';

    test('Should match the snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('Should display the image with the URL and default ALT', () => {
        render(<GifGridItem title={title} url={url} />)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Should display title in the component', () => {
        render(<GifGridItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    })

})