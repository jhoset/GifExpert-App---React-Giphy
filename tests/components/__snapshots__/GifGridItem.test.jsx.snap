// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> Should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hamlechi"
      src="https://hamlechi/photo.jpg"
    />
    <p>
       
      Hamlechi
       
    </p>
  </div>
</div>
`;
